### PR TITLE
Airneis 35: Init Config

### PR DESCRIPTION
**Pull request by cedricgautier**
# Summary
In this MR, we'll have to load the configuration for our application. 

We separated production and the staging environment, here by changing the field `build_production` to `true` you can choose to build the config with production variables. While it's false you can build with staging variables. 

By design, when production is loaded, staging variables are nil, and the same goes for the other way when our staging environment is loaded you may not load production variables.

## How to test : 
1. Change `build_production` to `true` if you want to load production variables in `config/default.yaml` file. 
2. Change in main.go, make these modifications to see the output of the field name

```go
	config, err := config.InitializeConfig()
	if err != nil {
		log.Fatal("Unable to load config - ", err.Error())
	}

	fmt.Println(config.Production.Env.Name)
	fmt.Println(config.Staging.Env.Name)
```
When it's production, you should have "Barnabe" as the output of the name field and "Cedric" when on staging. 

Screenshot 2024-01-23 at 23 18 56

As you can see there is an empty line on the screenshot, as proof when loading staging env the production variables aren't loaded and the field is nil as there is nothing inside.

### CODE REVIEW

**Review by KyDev25 — APPROVED** [review]

**Review by BarnabePILLIAUDIN — APPROVED** [review]
LGTM!

File: back/config/config.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		if err != nil {
+			return Config{}, errors.Wrapf(err, "InitializeConfig, unable to load staging config")
+		}
+		logrus.Info("Staging Environment has been loaded")
+	} else {
+		config, err = loadConfig(config, productionEnv)
+		if err != nil {
+			return Config{}, errors.Wrapf(err, "InitializeConfig, unable to load config")
+		}
+		logrus.Info("Production Environment has been loaded")
```
Comment (avetisk) [inline back/config/config.go:85]: Inconsistent style.

Comment (avetisk) [reply back/config/config.go:85]: Plus why are you not using early returns?

File: back/config/config.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+	default:
+		if err = yaml.Unmarshal(configFile, &productionConfig); err != nil {
+			return Config{}, errors.Wrapf(err, "loadConfig, unable to unmarshal")
+		}
+		config.Production = productionConfig
+	}
+
+	return config, nil
+}
```
Comment (avetisk) [inline back/config/config.go:115]: You should refactor this. There's clearly a repeating pattern.

**Pull request merged**